src/core/pratt.ml

```diff
@@ -0,0 +1,120 @@
+(** A simple parser for operators. See Pratt parsing or the Shunting Yard
+    algorithm.
+
+    <https://dev.to/jrop/pratt-parsing>
+    <https://effbot.org/zone/simple-top-down-parsing.htm> *)
+
+open Lplib
+open Lplib.Extra
+
+open Syntax
+
+(** Table containing defined binary operators. *)
+let bin_operators : binop StrHtbl.t = StrHtbl.create 17
+
+(** Table containing defined unary operators. *)
+let una_operators : unop StrHtbl.t = StrHtbl.create 17
+
+module Pratt : sig
+  val expression : ?rbp:priority -> p_term Stream.t -> p_term
```

Comment:
What rbp stand for?

Reply:
`rbp` stands for `right binding priority`. These are terms introduced in the Pratt parsing algorithm.

Reply:
Ok. It's good to define such acronyms in a comment.